packages/common/src/services/audius-backend/AudiusBackend.ts

```diff
@@ -1250,55 +1249,120 @@ export const audiusBackend = ({
     return receipt
   }
 
-  async function getTokenAccount({
+  /** Gets associated token account info for the passed account. It will
+   * first check if `address` ia a token account. If not, it will assume
+   * it is a root account and attempt to derive an associated token account from it.
+   */
+  async function getAssociatedTokenAccountInfo({
     address,
     sdk
   }: {
-    address: PublicKey
+    address: string
     sdk: AudiusSdk
   }) {
     const connection = sdk.services.solanaClient.connection
+    const pubkey = new PublicKey(address)
     try {
-      const tokenAccountInfo = await getAccount(connection, address)
-      return tokenAccountInfo
+      return await getAccount(connection, pubkey)
     } catch (err) {
+      // Account exists but is not a token account. Assume it's a root account
+      // and try to derive an associated account from it.
       if (err instanceof TokenInvalidAccountOwnerError) {
-        return null
+        console.info(
+          'Provided recipient solana address was not a token account. Assuming root account.'
+        )
+        const associatedTokenAccount = findAssociatedTokenAddress({
+          solanaWalletKey: pubkey,
+          mint: 'wAUDIO'
+        })
+        return await getAccount(connection, associatedTokenAccount)
       }
+      // Other error (including non-existent account)
       throw err
     }
   }
 
-  async function getAssociatedTokenAccountInfo({
+  async function createAssociatedTokenAccountWithPhantom(
+    connection: Connection,
+    address: string
+  ) {
+    if (!window.phantom) {
+      throw new Error(
+        'Recipient has no $AUDIO token account. Please install Phantom-Wallet to create one.'
+      )
+    }
+
+    if (!window.solana.isConnected) {
```

Comment:
what about mobile?

Reply:
Good question. This code didn't originally seem to handle it and I need to go dig and see how we were doing this on mobile before.

Reply:
We don't allow sending AUDIO on mobile, so I think this is covered for now.

Reply:
don't we need to check if window is defined?

Reply:
Talked offline, this is web desktop only code that was ported. Going to leave as-is for now.